andi/typeutils.py

```diff
@@ -151,3 +161,20 @@ def get_callable_func_obj(class_or_func: Callable) -> Callable:
         else:
             # not sure how to trigger it
             raise TypeError("Unexpected callable object %r" % (class_or_func,))
+
+
+def is_typing_annotated(o: Callable) -> bool:
+    """Returns True if the input is typing.Annotated and Python is 3.9+"""
+    if sys.version_info < (3, 9):
+        return False
+    from typing import Annotated, get_origin
+    return get_origin(o) == Annotated
+
+
+def strip_annotated(o: Callable) -> Callable:
```

Comment:
It's get called several times for each argument so we could do some caching here I guess.

The alternative would be saving the stripped results into vars but that makes the code more complicated.